pkg/kapp/diff/explicit_versioned_ref.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"fmt"
+)
+
+type VersionedRefDesc struct {
+	Namespace string `json:"namespace"`
+	APIGroup  string `json:"apiGroup"`
+	Kind      string `json:"kind"`
+	Name      string `json:"name"`
+}
+
+type ExplicitVersionedRefAnn struct {
+	References     []VersionedRefDesc `json:"references"`
+	VersionedNames map[string]string
+}
+
+type ExplicitVersionedRef struct {
+	Resource   VersionedResource
+	Annotation ExplicitVersionedRefAnn
+}
+
+const (
+	explicitReferenceKey = "kapp.k14s.io/versioned-explicit-ref"
```

Comment:
Nit: Move it to Versioned Resource file as it is used in that file only.

Reply:
The idea was to keep anything being used to implement this annotation in one file. I would say we should move this [function](https://github.com/vmware-tanzu/carvel-kapp/pull/311/files#diff-1299fda59a62e741c24d715184df939466614418c142b1dbf81cf9fc8840590aR118) to this file as well. Thoughts?

Reply:
No, that function should stay there as it belong to VersionedResource struct.

Reply:
It was added with this change so maybe should be a method of ExplicitVersionedRef struct instead